src/Microsoft.ML.Api/InternalSchemaDefinition.cs

```diff
@@ -16,7 +16,7 @@ namespace Microsoft.ML.Runtime.Api
     /// <summary>
     /// An internal class that holds the (already validated) mapping between a custom type and an IDataView schema.
     /// </summary>
-    internal sealed class InternalSchemaDefinition
+    public sealed class InternalSchemaDefinition
```

Comment:
>public sealed class InternalSchemaDefinition [](start = 4, length = 44)

I have similar concerns about this class being public. If we can avoid it, possibly by moving some of the logic current in `MemoryCollection.cs` to this assembly, then calling *that* from the `Microsoft.ML` project, I'd be delighted.